allocations/allocation_mapper.py

```diff
@@ -11,45 +34,129 @@ def _wants_db_allocations(self, request):
             return True
         return False
     
-    def map(self, project):
+    def map(self, project, get_redis_data = True):
         if self.allocations_from_db:
+            is_dict_type = isinstance(project, dict)
             balance_service = BalanceServiceClient()
             reformated_allocations = []
-            for alloc in portal_alloc.objects.filter(project_charge_code=project.chargeCode):
-                if alloc.status == 'active':
-                    active_allocation = balance_service.call(project.chargeCode)
+            project_charge_code = project['chargeCode'] if is_dict_type else project.chargeCode
+            for alloc in portal_alloc.objects.filter(project_charge_code=project_charge_code):
+                if get_redis_data and alloc.status == 'active':
+                    active_allocation = balance_service.call(project_charge_code)
                     if 'used' in active_allocation and active_allocation['used']:
-                        alloc.su_used = float(balance_service.call(project.chargeCode)['used'])
-                reformated_allocations.append(self.portal_to_tas_obj(alloc))
-            project.allocations = reformated_allocations
+                        alloc.su_used = float(balance_service.call(project_charge_code)['used'])
+                reformated_alloc = self.portal_to_tas_obj(alloc)
+                if is_dict_type:
```

Comment:
This is not for `__dict__`. There are several places that use this function, and depends on what `tasclient` method is used, it may have "project" as pytas.project object or a dict type.